### PR TITLE
Integration tests for GraphQL were created

### PR DESCRIPTION
**Pull request by vpetrovaa**
## PR-Codex overview
This PR adds GraphQL queries and mutations to retrieve, create, update, and delete notes. It also includes changes to the Note class and the schema file.

### Detailed summary
- Added GraphQL queries for retrieving notes by ID, user ID, and all notes.
- Added GraphQL mutations for creating, updating, and deleting notes.
- Changed the Note class to include a no-argument constructor.
- Updated the schema file to use Int instead of ID for note and input note IDs.
- Updated the pom.xml file to include dependencies for GraphQL and Testcontainers.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by vpetrovaa** [comment]
@h1alexbel is it a good practise for GraphQL integration tests to do like this:
Test to verify that we deleted the entity from db:
1. Create the entity and get this object
2. Get received id
3. Delete by this id
4. Call the existById method and assert false result
And all this steps are in one test?
So on the one hand all tests don't depend of each other and data that already exists in db, but on the other hand they become very complex

File: pom.xml [hunk 1]
```diff
@@ -119,6 +124,47 @@
...
+			<groupId>org.springframework.graphql</groupId>
+			<artifactId>spring-graphql-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-webflux</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
```
Comment (h1alexbel) [inline pom.xml:135]: compile dependencies should be separated with test one

**Review by h1alexbel — CHANGES_REQUESTED** [review]
@vpetrovaa check my comments

**Comment by h1alexbel** [comment]
> @h1alexbel is it a good practise for GraphQL integration tests to do like this: Test to verify that we deleted the entity from db:
> 
> 1. Create the entity and get this object
> 2. Get received id
> 3. Delete by this id
> 4. Call the existById method and assert false result
>    And all this steps are in one test?
>    So on the one hand all tests don't depend of each other and data that already exists in db, but on the other hand they become very complex

@vpetrovaa 
You should start the container in some class, that will be a parent for all other tests with database support for integration testing
then execute some logic, assert it inside your test
and then the method (in the parent class), for container destruction should be invoked

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**